### PR TITLE
Test the unpackSDKMeta with real data

### PR DESCRIPTION
**Pull request by borodovisin**
### Description
Create a unit test case with real data information from the logs to make sure the `unpackSDKMeta` is working as expected.

### Why are we making these changes?
To test the `unpackSDKMeta` with real information from merchants' use cases.

### CODE REVIEW

File: test/server/meta.test.js [hunk 1]
```diff
@@ -1021,3 +1021,60 @@ test('should error when invalid characters are found in the subdomain - we allow
...
+        'url':             'https://www.paypal.com/sdk/js?client-id=AfjD8CKlCiagY3OXl46F4HJKHIP7k2wRU3UDfMSuo52DBiXuc0srow-uyWGeHT0EXd67zADzzkm-5eoy&components=buttons&currency=EUR&intent=authorize&locale=de_DE&merchant-id=3PA5M5QDS5BES',
+        'attrs': {
+            'data-amount': '29.90',
+            'data-uid':    'uid_ssqwuttcumomcdjzbomwiakaomcjzs'
+        }
+    }
+];
+
+// $FlowIgnore[prop-missing] missing each property for test
+test.each(dataSet)('should unpack the sdk meta with same URL', (source) => {
```
Comment (gregjopa) [inline test/server/meta.test.js:1071]:
```suggestion
test.each(dataSet)('should unpack `sdkMeta` and load the script', (source) => {
```

File: test/server/meta.test.js [hunk 2]
```diff
@@ -1021,3 +1021,60 @@ test('should error when invalid characters are found in the subdomain - we allow
         throw new Error(`Expected error to be thrown`);
     }
 });
+
+const dataSet = [
```
Comment (gregjopa) [inline test/server/meta.test.js:1025]: What do you think about moving this `dataSet` and test to a separate file? Perhaps we create server/meta.integration.test.js

Comment (borodovisin) [reply test/server/meta.test.js:1025]: Sure that sounds good...

File: test/server/meta.integration.test.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/* @flow */
+
+import cheerio from 'cheerio';
+
+import { unpackSDKMeta } from '../../server';
+
+const sdkMetaList = [
```
Comment (westeezy) [inline test/server/meta.integration.test.js:7]: We can potentially simplify the building of theses so its easier to parse when reviewing tests. Right now we would have to really pause and read through each URL which can be difficult and potentially cause duplicate test cases or missed test cases in the long term. How do you feel about something like:
```
const params = [
  [
    {
      clientId: 'AQbghYd-7mRPyimEriYScIgTnYUsLnr5wVnPnmfPaSzwKrUe3qNzfEc5hXr9Ucf_JG_HFAZpJMJYXMuk',
      merchantId: 'K2US3FE3PQUEQ'
      intent: 'capture',
      lcaol: 'it_IT',
      components: ['buttons', 'funding-eligibility', 'messages']
      commit: false,
      currency: 'EUR'
    },
    {
      'data-partner-attribution-id': 'WooThemes_EC',
      'data-uid': 'uid_gwakesmdhfvakwylnuczrbplzdthxe',
    }
  ],
]

const buildTestUrl = ([queryParams, attrs]) => {
   // Build and return the url
  return `todo....`;
}

// $FlowIgnore[prop-missing] missing each property for test
test.each(params)('should unpack `sdkMeta` and load the script', ([queryParams, attrs]) => {
    const source = buildTestUrl([queryParams, attrs]);
    
    const { getSDKLoader } = unpackSDKMeta(Buffer.from(JSON.stringify(source))
        .toString('base64'));

    const $ = cheerio.load(getSDKLoader());
    const src = $('script').attr('src');

    if (source.url !== src) {
        throw new Error('The url should be the same');
    }
});
```

Comment (westeezy) [reply test/server/meta.integration.test.js:7]: doesn't have to be tuples but i jumped there. feel free to use objects of course.

Comment (westeezy) [reply test/server/meta.integration.test.js:7]: One could also make use of the test tables in jest but i kinda find those less readable than above to be honest

Comment (borodovisin) [reply test/server/meta.integration.test.js:7]: That sounds good @westeezy, you're right it's a little hard to understand the object.

**Review by wsbrunson — APPROVED** [review]

File: test/server/meta.integration.test.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+];
+
+/**
+ * Generate the URL from the source object
+ *
+ * @param {object} sourceData to generate the URL
+ * @returns {string} the generated URL including all the query parameters
+ */
+function buildUrl(sourceData) : string {
+    const { url, ...queryParam } = sourceData;
```
Comment (gregjopa) [inline test/server/meta.integration.test.js:107]:
```suggestion
    const { url, ...queryParams } = sourceData;
```

File: test/server/meta.integration.test.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+/**
+ * Generate the URL from the source object
+ *
+ * @param {object} sourceData to generate the URL
+ * @returns {string} the generated URL including all the query parameters
+ */
+function buildUrl(sourceData) : string {
+    const { url, ...queryParams } = sourceData;
+    const stringQueryParams = Object
```
Comment (gregjopa) [inline test/server/meta.integration.test.js:108]: Can you leverage `URLSearchParams` to simplify converting an object to a querystring?
Screen Shot 2022-02-17 at 2 27 55 PM

Comment (borodovisin) [reply test/server/meta.integration.test.js]: Sure @gregjopa, I'll make the changes

**Review by gregjopa — APPROVED** [review]
💯

**Pull request merged**